File: py/makeGood.py
class Solution:
    def makeGood(self, s: str) -> str:
        s_list = list(s)

        i = 1
        while i < len(s_list):
            if s_list[i].lower() == s_list[i - 1].lower() and s_list[i] != s_list[i - 1]:
                s_list.pop(i)
                s_list.pop(i - 1)
                i = 1
            else:
                i += 1
        
        return "".join(s_list)
            

File: cpp/makeGood.cpp
#include <stack>
#include <cctype>
#include <string>
using namespace std;
class Solution {
public:
    string makeGood(string s) {
        if (s.length() < 2) {return s;}
        stack<char> st;
        st.push(s[0]);

        for (int i = 1; i < s.length(); i ++) {
            if (!st.empty() && tolower(s[i]) == tolower(st.top()) && st.top() != s[i]) {
                st.pop();
            }
            else {
                st.push(s[i]);
            }
        }    
        
        string r;
        while (!st.empty()) {
            r = st.top() + r;
            st.pop();
        }
        return r;
    }
};